src/plugins/accounting.cpp

```diff
@@ -35,3 +35,85 @@ Association* get_association (int userid,
 
     return &bank_it->second;
 }
+
+
+json_t* Association::to_json () const
+{
+    json_t *held_job_ids = json_array ();
+    if (!held_job_ids) {
+        return nullptr;
+    }
+    for (const auto &job_id : held_jobs) {
+        json_array_append_new (held_job_ids, json_integer (job_id));
+    }
+
+    json_t *user_queues = json_array ();
+    if (!user_queues) {
+        json_decref (held_job_ids);
+        return nullptr;
+    }
+    for (const auto &queue : queues) {
+        json_array_append_new (user_queues, json_string (queue.c_str ()));
```

Comment:
I think technically the `json_string` call and the `json_array_append_new` should be checked. Like maybe:

```C++

json_t *temp;
if (!(temp = json_string (queue.c_str ())
    || json_array_append_new ( ... ) < 0 ){
return nullptr;
}

```

Reply:
There's a similar situation a few lines above.

Reply:
Oh neat! I didn't know you could combine checks like this in a single line - that's clever. I'll make these changes